src/python/ensembl/io/genomio/gff3/restructure.py

```diff
@@ -0,0 +1,264 @@
+    counts = _get_feat_counts(gene)
+    if counts.get("CDS") or counts.get("exon"):
+        raise GFFParserError(f"Gene {gene.id} contains direct CDSs and exons children")
+
+
+def add_transcript_to_naked_gene(gene: SeqFeature) -> None:
+    """Add an unspecific transcript to a gene without any sub features."""
+
+    if (len(gene.sub_features) > 0) or (gene.type != "gene"):
+        return
+
+    transcript = SeqFeature(gene.location, type="transcript")
+    transcript.qualifiers["source"] = gene.qualifiers["source"]
+    transcript.sub_features = []
+    gene.sub_features = [transcript]
+    logging.debug(f"Inserted 1 transcript for a lone gene {gene.id}")
+
+
+def move_only_cdss_to_new_mrna(gene: SeqFeature) -> None:
+    """Add intermediate mRNAs to a gene with only CDS children.
+    Do nothing if some sub-features are not CDS.
+    """
+
+    counts = _get_feat_counts(gene)
+    if (len(counts) != 1) or not counts.get("CDS"):
+        return
+
+    transcripts_dict = {}
+
+    for cds in gene.sub_features:
+        # We create as many transcripts as there are different CDS IDs
+        if cds.id not in transcripts_dict:
+            logging.debug(f"Create a new mRNA for {cds.id}")
+            transcript = SeqFeature(gene.location, type="mRNA")
+            transcript.qualifiers["source"] = gene.qualifiers["source"]
+            transcript.sub_features = []
+            transcripts_dict[cds.id] = transcript
+
+        # Add the CDS to the transcript
+        transcripts_dict[cds.id].sub_features.append(cds)
+
+        # Also add an exon in the same location
+        exon = SeqFeature(cds.location, type="exon")
+        exon.qualifiers["source"] = gene.qualifiers["source"]
+        exon.sub_features = []
+        transcripts_dict[cds.id].sub_features.append(exon)
+
+    transcripts = list(transcripts_dict.values())
+    gene.sub_features = transcripts
+
+    logging.debug(f"Insert transcript-exon feats for {gene.id} ({len(transcripts)} CDSs)")
+
+
+def move_only_exons_to_new_mrna(gene: SeqFeature) -> None:
+    """Add an mRNA for a gene that only has exons and move the exons under the mRNA.
+    No change if the gene has other sub_features than exon.
+    """
+
+    counts = _get_feat_counts(gene)
+    if (len(counts) != 1) or not counts.get("exon"):
+        return
+
+    transcript = SeqFeature(gene.location, type="mRNA")
+    transcript.qualifiers["source"] = gene.qualifiers["source"]
+    transcript.sub_features = gene.sub_features
+    gene.sub_features = [transcript]
+
+    logging.debug(f"Insert transcript for {gene.id} ({len(gene.sub_features)} exons)")
+
+
+def move_cds_to_existing_mrna(gene: SeqFeature) -> None:
+    """Move CDS child features of a gene, to the mRNA.
+
+    This is to fix the case where we have the following structure::
+        gene -> [ mRNA, CDSs ]
+    and change it to::
+        gene -> [ mRNA -> [ CDSs ] ]
+    The mRNA itself might have exons, in which case check that they match the CDS coordinates.
```

Comment:
```suggestion
    This is to fix the case where we have the following structure::
        gene -> [ mRNA, CDSs ]

    and change it to::
        gene -> [ mRNA -> [ CDSs ] ]

    The mRNA itself might have exons, in which case check that they match the CDS coordinates.
```
I'm no sure if to end a `::` block there is a need for an empty new line afterwards.